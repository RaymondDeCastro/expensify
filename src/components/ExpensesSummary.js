import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import getExpensesTotal from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';


export const ExpensesSummary = ({ expenseCount, totalAmount }) => (
  <div>
    <h1>
      {
        expenseCount === 1 ? 
        `Viewing 1 expense totalling ${numeral(totalAmount / 100).format('$0,0.00')}` :
        `Viewing ${expenseCount} expenses totalling ${numeral(totalAmount / 100).format('$0,0.00')}`
      }
    </h1>  
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    totalAmount: getExpensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary); 

