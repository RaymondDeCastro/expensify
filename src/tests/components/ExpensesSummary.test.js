import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('should create component for multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} totalAmount={132241}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should create component for 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} totalAmount={1341}/>);
  expect(wrapper).toMatchSnapshot();
});