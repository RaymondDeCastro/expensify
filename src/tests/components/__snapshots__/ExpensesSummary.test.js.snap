// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create component for 1 expense 1`] = `
<div>
  <h1>
    Viewing 1 expense totalling $13.41
  </h1>
</div>
`;

exports[`should create component for multiple expenses 1`] = `
<div>
  <h1>
    Viewing 3 expenses totalling $1,322.41
  </h1>
</div>
`;
